### PR TITLE
Async load fix

### LINKED ISSUE #3500: bokeh.js must load first, before widgets, compiler
refererence: https://groups.google.com/a/continuum.io/forum/#!topic/bokeh/aJPeJUottKk

ping @mattpap 

Any immediate thoughts? There are definitely ways to chain callbacks together to load one first. We'd have to special case the first to be the "load first" script, or else chain them all together in order in general. Or we could put a fixed number of timeout retries when `Bokeh` is not found as a stopgap.

### PR DESCRIPTION
**Pull request by havocp**
An attempt to fix #3500 

This is kind of a risky change. I changed how Bokeh works always, not only for notebooks, because it ended up seeming trickier to think about doing it only for notebooks and I was less confident in that.

Supposed to work in IE10 and above: http://caniuse.com/#feat=script-async

The essence of the change is that now you should NEVER assume "Bokeh" exists unless you are wrapped in `bokeh_load` like `bokeh_load(function() { Bokeh.foo })`.

I just realized I didn't test the autoload stuff yet, so it's probably hosed. Let's get travis going on this as-is though.

### CODE REVIEW

**Comment by mattpap** [comment]
btw. I'm working on a different approach where you don't have to manually wrap every piece of javascript we produce.

**Comment by havocp** [comment]
Sounds good. Check my last commit, don't know if you're talking about the same thing - I was autowrapping the js_raw but then it wraps the inline bokeh.js itself - which actually works but it's kind of strange. Seemed clearer to show the full final script in one place.

**Comment by bryevdv** [comment]
The S3 report actually looks fairly good, I think it is mainly just unit tests. There a a few `charts/file` examples that did not render, I asked @Maggie-M to check out the branch and try to run those locally. As an aside all the notebooks rendered in the report I think, I wonder if this explains why notebooks sometimes failed to render in the tests. I'll take a look over the PR code as well.

File: bokeh/core/_templates/js_resources.html [hunk 1]
```diff
@@ -2,16 +2,17 @@
...
-:param js_raw: a list of raw JS snippets to put between ``<style>`` tags
+:param js_raw: a list of raw JS snippets to put between ``<script>`` tags
 :type js_raw: list[str]
 
 #}
-{%- for file in js_files %}
-<script type="text/javascript" src="{{ file }}"></script>
-{%- endfor %}
+
+<script type="text/javascript"">
```
Comment (bryevdv) [inline bokeh/core/_templates/js_resources.html]: extra `"`

**Comment by havocp** [comment]
doh, unit tests. fixing.

**Comment by bryevdv** [comment]
`bokeh/examples/embed/animated.py` does not seem to work. `autoload.js` returns this script:
```
(function(global) {

  

  var elt = document.getElementById("7b55a535-bf71-435e-ae13-f5b4fbade24b");
  if(elt==null) {
    console.log("Bokeh: ERROR: autoload.js configured with elementid '7b55a535-bf71-435e-ae13-f5b4fbade24b' but no matching script tag was found. ")
    return false;
  }

  var websocket_url = "ws://localhost:5006/ws";

  // Docs will be set for the static case
  var docs_json = {};

  var render_items = [{ 'elementid' : "7b55a535-bf71-435e-ae13-f5b4fbade24b",
                        'sessionid' : "BxGQmnoNHn1Rqjs9mAeDKE4WEO9DJotQmioibPXVZ0G9" }];

  bokeh_load(function() {
    Bokeh.embed.embed_items(docs_json, render_items, websocket_url);
  });

}(this));
``` which results in `bokeh_load` undefined error

**Comment by bryevdv** [comment]
@havocp @mattpap I am going to close this PR. I am going to start a new PR to fix the notebook comms problems but I would like to investigate some of the merits of this approach in that work.

**Pull request closed without merging**